code/game/machinery/doors/firedoor.dm

```diff
@@ -203,30 +198,59 @@
 		for(var/obj/machinery/firealarm/fire_panel in place.firealarms)
 			fire_panel.set_status()
 
-/obj/machinery/door/firedoor/proc/check_atmos(datum/source)
+/obj/machinery/door/firedoor/proc/refresh_shared_turfs(datum/source, list/leaving_members, list/joining_members)
+	SIGNAL_HANDLER
+	
+	for(var/obj/machinery/door/firedoor/firelock in my_area.firedoors)
+		firelock.check_adjacent_turfs()
+
+/obj/machinery/door/firedoor/proc/check_adjacent_turfs()
+	for(var/dir in GLOB.cardinals)
+		var/turf/checked_turf = get_step(get_turf(src),dir)
+		if(checked_turf && !checked_turf.density)
```

Comment:
we don't care about density here, we want to register onto walls too

Reply:
wont let me resolve this